projects/packages/jetpack-mu-wpcom/src/features/wpcom-media/wpcom-media-url-upload.php

```diff
@@ -76,7 +84,25 @@ function wpcom_handle_media_url_upload() {
 	}
 }
 
-if ( current_user_can( 'upload_files' ) ) {
-	add_action( 'pre-upload-ui', 'wpcom_media_url_upload', 9 );
+/**
+ * Load the wpcom media URL upload form.
+ */
+function load_wpcom_media_url_upload_form() {
+	if ( ! current_user_can( 'upload_files' ) ) {
+		return;
+	}
+
 	add_action( 'wp_ajax_wpcom_media_url_upload', 'wpcom_handle_media_url_upload' );
+
+	if ( empty( $_GET['untangling-media'] ) ) { // phpcs:disable WordPress.Security.NonceVerification.Recommended
+		return;
+	}
+
+	global $pagenow;
+	if ( $pagenow !== 'media-new.php' ) {
+		add_action( 'pre-upload-ui', 'append_wpcom_media_url_upload', 9 );
+	}
+
+	add_action( 'post-plupload-upload-ui', 'enqueue_wpcom_media_url_upload_form' );
```

Comment:
Should this be inside `else`? Because for example in `upload.php`, we don't need this action. So something like this

```php
	if ( $pagenow === 'media-new.php' ) {
        add_action( 'post-plupload-upload-ui', 'enqueue_wpcom_media_url_upload_form' );
	} else {
		add_action( 'pre-upload-ui', 'append_wpcom_media_url_upload', 9 );
    }
```

Reply:
We need this action to enqueue js to both `media-new.php` and `upload.php`. Or do we mean we can simply call `enqueue_wpcom_media_url_upload_form` inside the `append_wpcom_media_url_upload` function?

Reply:
Ooops sorry I misread the code, you're right 🤦‍♂️